core/src/test/java/org/testcontainers/containers/ReportLeakedContainers.java

```diff
@@ -0,0 +1,44 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.model.Container;
+import org.testcontainers.DockerClientFactory;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonMap;
+import static java.util.stream.Collectors.joining;
+
+public class ReportLeakedContainers
+{
+    public static Optional<String> reportLeakedContainers()
```

Comment:
because of the global/static nature of SESSION_ID this method should be used in a single-threaded manner. `synchronized` + `callable` to check.

Reply:
It's not possible to control other container creators, so this test has to run exclusively. 
However we are safe here because: `JUnit itself does not provide support for parallel test execution`